Initial commit for NCAlgebra package to Macaulay2 repository.

moorewf opened the pull request.

DanGrayson commented:
I tried installing the package (with bergman 1.001 installed) and it tried to rerun one of the examples, but failed.  If I try again with `RerunExamples => true` then I get 8 errors.  If I analyze one of them, I get this:
```
einsteinium$ bergman -i /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/5.init -on-error exit --silent
*** We turn on noncommutativity

NIL 
T 
NIL 
NIL algebraic form input> 
T 
#\X *** The file  /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/7.bgb exists. Shall I try to remove it? 
*** - FUNCALL: undefined function YESP

einsteinium$ cat /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/5.init
(setf (getenv "bmload") "/lap/clisp/unix")
(ncpbhgroebner "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/6.bi" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/7.bgb" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/9.bpb" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/8.bhs")
(calctolimit (getmaxdeg))
(tdegreehseriesout (getmaxdeg))
(quit)
```

Comments?

moorewf commented:
Not sure what the "YESP" function is.  Getting bergman to work under the
most recent Mac OS X is a bit difficult.  Did you follow instructions in
the installBergman.pdf file in the auxiliary files directory?

If so, could you send me the temp files (specifically 5.init and 6.bi) so I
can try and track it down?  Also, do you know what documentation node it
was generating when it had an error?

Thanks,

Frank

On Sun, Dec 17, 2017 at 6:23 PM, Daniel R. Grayson <notifications@github.com
> wrote:

> I tried installing the package (with bergman 1.001 installed) and it tried
> to rerun one of the examples, but failed. If I try again with RerunExamples
> => true then I get 8 errors. If I analyze one of them, I get this:
>
> einsteinium$ bergman -i /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/5.init -on-error exit --silent
> *** We turn on noncommutativity
>
> NIL
> T
> NIL
> NIL algebraic form input>
> T
> #\X *** The file  /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/7.bgb exists. Shall I try to remove it?
> *** - FUNCALL: undefined function YESP
>
> einsteinium$ cat /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/5.init
> (setf (getenv "bmload") "/lap/clisp/unix")
> (ncpbhgroebner "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/6.bi" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/7.bgb" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/9.bpb" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23364-0/8.bhs")
> (calctolimit (getmaxdeg))
> (tdegreehseriesout (getmaxdeg))
> (quit)
>
> Comments?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Macaulay2/M2/pull/720#issuecomment-352294379>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFj957wQSQXoNnEDNLv0Dit0JXK6Wbjjks5tBaKFgaJpZM4RE0Gx>
> .
>



-- 
Dr. W. Frank Moore
Associate Professor
Graduate Recruiting Chair
Department of Mathematics and Statistics
Wake Forest University

email: moorewf@wfu.edu

mikestillman commented:
The installation instructions work, I think, for Sierra (macOS 10.12.6).  All of the examples run to completion.

DanGrayson commented:
More detail below.  Notice that the file 10.ter has a more interesting
error message: `*** - LOAD: A file with name /lap/clisp/unix/hseries.fas does not exist`.
But it does exist, at least if you add the right prefix.


```
einsteinium$ bergman -i /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/5.init -on-error exit --silent
*** We turn on noncommutativity

NIL 
T 
NIL 
NIL algebraic form input> 
T 
#\X *** The file  /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/7.bgb exists. Shall I try to remove it? 
*** - FUNCALL: undefined function YESP

einsteinium$ head -100 /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/*
==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/0.init <==
(simple "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/1.bi" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/2.bo")
(quit)

==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/1.bi <==
(noncommify)
(setmodulus 0)
(setweights 1 1 1 )
(setmaxdeg 5)
(algforminput)
vars x,y,z;
z*y+y*z-x^2,
x*z+z*x-y^2,
y*x+x*y-z^2;


==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/10.ter <==
*** We turn on noncommutativity

NIL 
T 
NIL 
NIL algebraic form input> 
T 
#\X 
*** - LOAD: A file with name /lap/clisp/unix/hseries.fas does not exist

==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/2.bo <==
% 2
z*x-y^2+x*z,
   z*y+y*z-x^2,
   -z^2+y*x+x*y,
   
% 3
y*x^2-x^2*y,
   -y^2*x+x*y^2,
   
Done

==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/3.ter <==

NIL 
T 
NIL 
NIL algebraic form input> 
T 
#\X  - All is OK (I hope). Now you may (e. g.):
   - kill bergman with (QUIT); or
   - interrupt bergman with ^Z; or
   - clear the memory with (CLEARIDEAL), and run a new (SIMPLE).

==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/4.bigb <==
(load "/lap/clisp/unix/hseries.fas")
(setinterruptstrategy minhilblimits)
(setinterruptstrategy minhilblimits)
(sethseriesminima skipcdeg skipcdeg skipcdeg skipcdeg skipcdeg skipcdeg)
(noncommify)
(setmodulus 0)
(setweights 1 1 1 )
(setmaxdeg 6)
(algforminput)
vars x,y,z;
% 2
z*x-y^2+x*z,
   z*y+y*z-x^2,
   -z^2+y*x+x*y,
   
% 3
y*x^2-x^2*y,
   -y^2*x+x*y^2;


==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/5.init <==
(setf (getenv "bmload") "/lap/clisp/unix")
(ncpbhgroebner "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/6.bi" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/7.bgb" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/9.bpb" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/8.bhs")
(calctolimit (getmaxdeg))
(tdegreehseriesout (getmaxdeg))
(quit)

==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/6.bi <==
(noncommify)
(setmodulus 0)
(setweights 1 1 1 )
(setmaxdeg 10)
(algforminput)
vars x,y,z;
z*y+y*z-x^2,
x*z+z*x-y^2,
y*x+x*y-z^2;


==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/7.bgb <==

==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/8.bhs <==

==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/9.bpb <==
einsteinium$ ls -l ~/src/bergman1.001/lap/clisp/unix/hseries.fas
-rw-r--r-- 1 dan staff 89736 Dec 17 17:11 /Users/dan/src/bergman1.001/lap/clisp/unix/hseries.fas
einsteinium$ type bergman
bergman is hashed (/Users/dan/bin/bergman)
einsteinium$ ls -l /Users/dan/bin/bergman
lrwxrwxr-x 1 dan staff 50 Dec 17 17:14 /Users/dan/bin/bergman -> /Users/dan/src/bergman1.001/bin/clisp/unix/bergman
```

moorewf commented:
That error should be fixed with updating the environment variable
BERGMANPATH to the directory containing your bergman installation.

This fixes a bug in the bergman distribution; for some reason the system
doesn't load the Hilbert series code by default and you have to tell it to
load the hseries.fas file (so I have to include that command in 4.bigb).

On Sun, Dec 17, 2017 at 8:43 PM, Daniel R. Grayson <notifications@github.com
> wrote:

> More detail below. Notice that the file 10.ter has a more interesting
> error message: *** - LOAD: A file with name /lap/clisp/unix/hseries.fas
> does not exist.
> But it does exist, at least if you add the right prefix.
>
> einsteinium$ bergman -i /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/5.init -on-error exit --silent
> *** We turn on noncommutativity
>
> NIL
> T
> NIL
> NIL algebraic form input>
> T
> #\X *** The file  /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/7.bgb exists. Shall I try to remove it?
> *** - FUNCALL: undefined function YESP
>
> einsteinium$ head -100 /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/*
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/0.init <==
> (simple "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/1.bi" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/2.bo")
> (quit)
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/1.bi <==
> (noncommify)
> (setmodulus 0)
> (setweights 1 1 1 )
> (setmaxdeg 5)
> (algforminput)
> vars x,y,z;
> z*y+y*z-x^2,
> x*z+z*x-y^2,
> y*x+x*y-z^2;
>
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/10.ter <==
> *** We turn on noncommutativity
>
> NIL
> T
> NIL
> NIL algebraic form input>
> T
> #\X
> *** - LOAD: A file with name /lap/clisp/unix/hseries.fas does not exist
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/2.bo <==
> % 2
> z*x-y^2+x*z,
>    z*y+y*z-x^2,
>    -z^2+y*x+x*y,
>
> % 3
> y*x^2-x^2*y,
>    -y^2*x+x*y^2,
>
> Done
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/3.ter <==
>
> NIL
> T
> NIL
> NIL algebraic form input>
> T
> #\X  - All is OK (I hope). Now you may (e. g.):
>    - kill bergman with (QUIT); or
>    - interrupt bergman with ^Z; or
>    - clear the memory with (CLEARIDEAL), and run a new (SIMPLE).
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/4.bigb <==
> (load "/lap/clisp/unix/hseries.fas")
> (setinterruptstrategy minhilblimits)
> (setinterruptstrategy minhilblimits)
> (sethseriesminima skipcdeg skipcdeg skipcdeg skipcdeg skipcdeg skipcdeg)
> (noncommify)
> (setmodulus 0)
> (setweights 1 1 1 )
> (setmaxdeg 6)
> (algforminput)
> vars x,y,z;
> % 2
> z*x-y^2+x*z,
>    z*y+y*z-x^2,
>    -z^2+y*x+x*y,
>
> % 3
> y*x^2-x^2*y,
>    -y^2*x+x*y^2;
>
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/5.init <==
> (setf (getenv "bmload") "/lap/clisp/unix")
> (ncpbhgroebner "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/6.bi" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/7.bgb" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/9.bpb" "/var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/8.bhs")
> (calctolimit (getmaxdeg))
> (tdegreehseriesout (getmaxdeg))
> (quit)
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/6.bi <==
> (noncommify)
> (setmodulus 0)
> (setweights 1 1 1 )
> (setmaxdeg 10)
> (algforminput)
> vars x,y,z;
> z*y+y*z-x^2,
> x*z+z*x-y^2,
> y*x+x*y-z^2;
>
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/7.bgb <==
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/8.bhs <==
>
> ==> /var/folders/46/9b86vqxj4hjcngvy7kd7sb140000gn/T/M2-23356-0/9.bpb <==
> einsteinium$ ls -l ~/src/bergman1.001/lap/clisp/unix/hseries.fas
> -rw-r--r-- 1 dan staff 89736 Dec 17 17:11 /Users/dan/src/bergman1.001/lap/clisp/unix/hseries.fas
> einsteinium$ type bergman
> bergman is hashed (/Users/dan/bin/bergman)
> einsteinium$ ls -l /Users/dan/bin/bergman
> lrwxrwxr-x 1 dan staff 50 Dec 17 17:14 /Users/dan/bin/bergman -> /Users/dan/src/bergman1.001/bin/clisp/unix/bergman
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Macaulay2/M2/pull/720#issuecomment-352305382>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFj952NvFZPXlF0SdKglWTurcO5joi0iks5tBcM9gaJpZM4RE0Gx>
> .
>



-- 
Dr. W. Frank Moore
Associate Professor
Graduate Recruiting Chair
Department of Mathematics and Statistics
Wake Forest University

email: moorewf@wfu.edu

mikestillman commented:
Maybe the pdf instructions should be changed to say that NCAlgebra now comes with M2 (i.e. doesn't need to be downloaded and installed?

mikestillman commented:
the build succeeds, no errors, all tests pass.  Are there tests for this package?  

@DanGrayson what is the best way to write tests for a package like this where the executable might not be present?  I know there have been some packages lately that do similar things.

DanGrayson commented:
We don't have an official way to segregate tests into two sorts depending on whether external programs are present, and the makefile, in response to `make check`, runs all the tests, so for the sake of our overnight build, there'd better be no tests that depend on bergman being there, or we have to install bergman on the overnight build machine.  Nothing prevents us from installing all the external program on the build machine, so alternatively, we could allow tests that depend on external components.

DanGrayson commented:
Thanks, Frank, I was ignoring the instruction about BERGMANPATH!  But instead of adding
it to my .profile, I made a little shell script called bergman that sets the variable and calls the real bergman.  (With Macaulay2, we were careful to avoid putting a requirement like that on our users -- M2 can find its files even if you run it via a symbolic link.)

How about (eventually) fixing your code so it locates the error message from bergman in the 10.ter file and displays it to the user as part of the error message?

DanGrayson commented:
It turns out that bergman knows well where its files are, so there is no need to set up the environment variable BERGMANPATH for the benefit of your package!  (Actually, if your package had really needed to know that, it should have gone in the the package's "configuration file" instead.)

I fixed a couple of other things, too:

Here are the diffs you need:
```
diff --git a/M2/Macaulay2/packages/NCAlgebra.m2 b/M2/Macaulay2/packages/NCAlgebra.m2
index 2c9cc302a..9d16e1a62 100644
--- a/M2/Macaulay2/packages/NCAlgebra.m2
+++ b/M2/Macaulay2/packages/NCAlgebra.m2
@@ -13,7 +13,6 @@ newPackage("NCAlgebra",
 	   HomePage => "http://people.hamilton.edu/cgibbons/index.html",
 	   Email => "crgibbon@hamilton.edu"}},
      AuxiliaryFiles => true,
-     DebuggingMode => true,
      CacheExampleOutput =>true
      )
 
@@ -96,12 +95,6 @@ protect CumulativeBasis
 MAXDEG = 40
 MAXSIZE = 1000
 
--- This manner of locating the bergman source code is deprecated.
--- bergmanPath = "~/bergman"
-
--- the environment variable BERGMANPATH must be set to the root directory of the bergman source
-bergmanPath = getenv "BERGMANPATH"
-
 NCRing = new Type of Ring
 NCQuotientRing = new Type of NCRing
 NCPolynomialRing = new Type of NCRing
@@ -1414,7 +1407,7 @@ writeBergmanInputFile (NCRing,String,String) := opts -> (B,genListString,tempInp
       -- if we don't want to recompute the GB, we need to tell Bergman that there are no
       -- Spairs to work on for twice the max degree of the gens we send it so it
       -- doesn't try to create any more Spairs.
-      fil << "(load \"" << bergmanPath << "/lap/clisp/unix/hseries.fas\")" << endl;
+      fil << "(load (mkbmpathexpand \"$bmload/lap/clisp/unix/hseries.fas\"))" << endl;
       fil << "(setinterruptstrategy minhilblimits)" << endl;
       fil << "(setinterruptstrategy minhilblimits)" << endl;
       fil << "(sethseriesminima" << concatenate(opts#DegreeLimit:" skipcdeg") << ")" << endl;
@@ -1676,7 +1669,7 @@ writeHSInitFile := (tempInit,
 		    tempPBOutput,
 		    tempHSOutput) -> (
    fil := openOut tempInit;
-   fil << "(setf (getenv \"bmload\") \"" << bergmanPath << "/lap/clisp/unix\")" << endl;
+   fil << "(setf (getenv \"bmload\") (mkbmpathexpand \"$bmload/lap/clisp/unix\"))" << endl;
    fil << "(ncpbhgroebner " 
        << "\"" << tempInput << "\" "
        << "\"" << tempGBOutput << "\" "
@@ -3542,7 +3535,7 @@ wallTiming = f -> (
 ------------------------------------------------------------
 
 --- include the documentation
-load (currentFileDirectory | "NCAlgebra/NCAlgebraDoc.m2")
+load "./NCAlgebra/NCAlgebraDoc.m2"
 
 end
 
```

moorewf commented:
Great!  I am working on fixing the installation instructions and the
changes will reflect this update as well.



On Mon, Dec 18, 2017 at 4:03 PM, Daniel R. Grayson <notifications@github.com
> wrote:

> It turns out that bergman knows well where its files are, so there is no
> need to set up the environment variable BERGMANPATH for the benefit of your
> package! (Actually, if your package had really needed to know that, it
> should have gone in the the package's "configuration file" instead.)
>
> I fixed a couple of other things, too:
>
> Here are the diffs you need:
>
> diff --git a/M2/Macaulay2/packages/NCAlgebra.m2 b/M2/Macaulay2/packages/NCAlgebra.m2
> index 2c9cc302a..9d16e1a62 100644
> --- a/M2/Macaulay2/packages/NCAlgebra.m2
> +++ b/M2/Macaulay2/packages/NCAlgebra.m2
> @@ -13,7 +13,6 @@ newPackage("NCAlgebra",
>  	   HomePage => "http://people.hamilton.edu/cgibbons/index.html",
>  	   Email => "crgibbon@hamilton.edu"}},
>       AuxiliaryFiles => true,
> -     DebuggingMode => true,
>       CacheExampleOutput =>true
>       )
>
> @@ -96,12 +95,6 @@ protect CumulativeBasis
>  MAXDEG = 40
>  MAXSIZE = 1000
>
> --- This manner of locating the bergman source code is deprecated.
> --- bergmanPath = "~/bergman"
> -
> --- the environment variable BERGMANPATH must be set to the root directory of the bergman source
> -bergmanPath = getenv "BERGMANPATH"
> -
>  NCRing = new Type of Ring
>  NCQuotientRing = new Type of NCRing
>  NCPolynomialRing = new Type of NCRing
> @@ -1414,7 +1407,7 @@ writeBergmanInputFile (NCRing,String,String) := opts -> (B,genListString,tempInp
>        -- if we don't want to recompute the GB, we need to tell Bergman that there are no
>        -- Spairs to work on for twice the max degree of the gens we send it so it
>        -- doesn't try to create any more Spairs.
> -      fil << "(load \"" << bergmanPath << "/lap/clisp/unix/hseries.fas\")" << endl;
> +      fil << "(load (mkbmpathexpand \"$bmload/lap/clisp/unix/hseries.fas\"))" << endl;
>        fil << "(setinterruptstrategy minhilblimits)" << endl;
>        fil << "(setinterruptstrategy minhilblimits)" << endl;
>        fil << "(sethseriesminima" << concatenate(opts#DegreeLimit:" skipcdeg") << ")" << endl;
> @@ -1676,7 +1669,7 @@ writeHSInitFile := (tempInit,
>  		    tempPBOutput,
>  		    tempHSOutput) -> (
>     fil := openOut tempInit;
> -   fil << "(setf (getenv \"bmload\") \"" << bergmanPath << "/lap/clisp/unix\")" << endl;
> +   fil << "(setf (getenv \"bmload\") (mkbmpathexpand \"$bmload/lap/clisp/unix\"))" << endl;
>     fil << "(ncpbhgroebner "
>         << "\"" << tempInput << "\" "
>         << "\"" << tempGBOutput << "\" "
> @@ -3542,7 +3535,7 @@ wallTiming = f -> (
>  ------------------------------------------------------------
>
>  --- include the documentation
> -load (currentFileDirectory | "NCAlgebra/NCAlgebraDoc.m2")
> +load "./NCAlgebra/NCAlgebraDoc.m2"
>
>  end
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Macaulay2/M2/pull/720#issuecomment-352556638>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFj9545nD_Q1LTAIEeIfEKRwSe4w9aeoks5tBtMHgaJpZM4RE0Gx>
> .
>



-- 
Dr. W. Frank Moore
Associate Professor
Graduate Recruiting Chair
Department of Mathematics and Statistics
Wake Forest University

email: moorewf@wfu.edu

moorewf commented:
If I want to update the install file to reflect the changes (that NCAlgebra is included in the M2 distribution, and also removing the fix regarding BERGMANPATH), how do I change this pull request, since it is based on an old state?

DanGrayson commented:
Updating a pull request is done simply by you pushing to your branch on which the pull request was made, as you have done.

DanGrayson commented:
The documentation file is a pdf file.  It would be good if you would also include the source TeX file from which it's made, so others can maintain your package (in the long run).

Alternatively, why not just have the file be a plain text file?

moorewf commented:
Good idea.  The TeX source has been added.

Do you think I should ditch the pdf file entirely and only use text?

DanGrayson commented:
Yes, why not ditch the pdf?  Wouldn't plain ascii suffice?

mikestillman commented:
I would recommend also making a link to it from your documentation front page for NCAlgebra package.  That way, when someone tries to use it, and it fails, they can see right away what to do to fix it.

> On Dec 19, 2017, at 1:02 PM, Daniel R. Grayson <notifications@github.com> wrote:
> 
> Yes, why not ditch the pdf? Wouldn't plain ascii suffice?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/Macaulay2/M2/pull/720#issuecomment-352838537>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAGPR-sSl4_SnGncjOMVo-jL0MaLFPiSks5tB_pNgaJpZM4RE0Gx>.
>

moorewf commented:
I have added the link in the documentation node (it currently points to where it will be in the M2/master branch, so the link is currently dead) and converted the instructions to a plain text file.

DanGrayson commented:
Let's eliminate BERGMANPATH and bergmanPath completely, as I demonstrated in my suggested patch.

DanGrayson commented:
Hi, I'd like to make version 1.11 soon.  What about that minor fix?

moorewf commented:
I tried last week to implement your changes.  The issue seems to be that
bergman doesn't know what $bmload is when the init script is run (your code
works fine from a bergman command line).

I am going to play with it some more today to see if I can get it to work
without the environment variable.

Frank

On Mon, Jan 15, 2018 at 8:02 PM, Daniel R. Grayson <notifications@github.com
> wrote:

> Hi, I'd like to make version 1.11 soon. What about that minor fix?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Macaulay2/M2/pull/720#issuecomment-357823716>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFj95y6FXo3wLw5IUGXpGvZ9ESvD_mI4ks5tK_UdgaJpZM4RE0Gx>
> .
>



-- 
Dr. W. Frank Moore
Associate Professor
Graduate Recruiting Chair
Department of Mathematics and Statistics
Wake Forest University

email: moorewf@wfu.edu

DanGrayson commented:
If you don't get it, tell me what "the init script" is, and I'll take a look.

moorewf commented:
The problem seems to be a bergman issue, not really with the package.

My M2 package calls the command:

bergman -i 5.init -on-error exit --silent > 10.ter

on the file 5.init, which references 6.bi (both of which are attached to
this email).  The 5.init file currently has your 'bmload' command, though
it has worked when in the 6.bi as well when expanding the environment
variable on the M2 end of things.

You can also remove the bmload command entirely, and it ends up having the
same error (although later on) - it seems that some part of the Hilbert
series code is having trouble finding the "hseries.fas" file.

Any help is greatly appreciated,

Frank

On Tue, Jan 16, 2018 at 10:42 AM, Daniel R. Grayson <
notifications@github.com> wrote:

> If you don't get it, tell me what "the init script" is, and I'll take a
> look.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Macaulay2/M2/pull/720#issuecomment-358003755>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFj951Z3G_ErelbrhxV82pRAWswxoGk1ks5tLMNqgaJpZM4RE0Gx>
> .
>



-- 
Dr. W. Frank Moore
Associate Professor
Graduate Recruiting Chair
Department of Mathematics and Statistics
Wake Forest University

email: moorewf@wfu.edu

DanGrayson commented:
I fixed it by applying the patch I provided before, and it works: merged.
